### PR TITLE
Adjust SMIRFF format spec to address generalization of format

### PR DESCRIPTION
**Pull request by davidlmobley**
- [x] Discuss how other mixing/combination rules would be handled
- [x] Discuss how other functional forms for the nonbonded term would be selected
- [x] Discuss how other functional forms for valence terms such as angle terms would be selected (and mention switch to `AngleForce` rather than `HarmonicAngleForce`, as per #179 
- [x] Remove id/parent_id from formal spec, just mentioning `id` in a usage example
- [x] Put in mention of GB parameters
- [x] Add line breaks for better change logs

@jchodera , anything else I'm missing?

### CODE REVIEW

File: The-SMIRFF-force-field-format.md [hunk 1]
```diff
@@ -1,33 +1,40 @@
...
+The SMIRFF format provides XML `ffxml` files that are parseable by the `ForceField` class of the `smarty.forcefield` module. 
+These encode parameters for a force field based on a SMIRKS-based specification of the chemical environment the parameters are to be applied to. 
+The file has tags corresponding to OpenMM force terms (`HarmonicBondForce`, `HarmonicAngleForce`, `PeriodicTorsionForce`, etc., as discussed in more detail below); these specify units used for the different constants provided for individual force terms, for example (see the [AlkEthOH example ffxml](https://github.com/open-forcefield-group/smarty/blob/master/smarty/data/forcefield/Frosst_AlkEtOH.ffxml)):
 ```XML
    <HarmonicAngleForce angle_unit="degrees" k_unit="kilocalories_per_mole/radian**2">
 ```       
 which introduces following `Angle` terms which will use units of degrees for the angle and kilocalories per mole per square radian for the force constant.
 
 Under each of these force terms, there are tags for individual parameter lines such as these:
 ```XML
```
Comment (jchodera) [inline The-SMIRFF-force-field-format.md:17]: I'd add the `<AngleForce>...</AngleForce>` tags here to make it very clear that the `<Angle.../>` tags have to go inside this block.

File: The-SMIRFF-force-field-format.md [hunk 2]
```diff
@@ -53,64 +60,74 @@ For compatibility, the size property of an atom can be specified either via prov
...
 ```
 
-**AMBER functional forms drop the factor of 2 in the angle energy term, which we elect not to do here.** Thus, comparing a SMIRFF file to a corresponding AMBER parameter file or .frcmod will make it appear that force constants here are twice as large as they ought to be.
+**AMBER functional forms drop the factor of 2 in the angle energy term, which we elect not to do here.**
+Thus, comparing a SMIRFF file to a corresponding AMBER parameter file or .frcmod will make it appear that force constants here are twice as large as they ought to be.
 
 ### PROPER TORSIONS
 
 Torsions are implemented as a [`PeriodicTorsionForce`](http://docs.openmm.org/7.0.0/userguide/theory.html#periodictorsionforce) tag with child tags for `Proper` (discussed here) and `Improper` (discussed below) parameters, for example:
 ```
```
Comment (jchodera) [inline The-SMIRFF-force-field-format.md]: Forgot `XML` after '```'

**Comment by jchodera** [comment]
Are you planning to change the `<Harmonic*Force.../>` tags to drop the `Harmonic` part in this PR?

**Comment by davidlmobley** [comment]
@jchodera :

> Are you planning to change the `<Harmonic*Force.../>` tags to drop the Harmonic part in this PR?

I was going to wait til we change the code to do that. Right now it's listed here under things we are GOING to do. When we do it, then I'll drop the `Harmonic`.

File: The-SMIRFF-force-field-format.md (hunk 2 above)
Comment (davidlmobley) [reply The-SMIRFF-force-field-format.md]: Thanks.

**Comment by jchodera** [comment]
How about this plan:
- Commit this PR once ready
- Create a new branch for the new updates (e.g. `0.2.0`) and make the updates to the spec
- I'll then update the code to match the spec
- We then merge the whole branch into `master`.

File: The-SMIRFF-force-field-format.md (hunk 1 above)
Comment (davidlmobley) [reply The-SMIRFF-force-field-format.md:17]: Wait, hold on -- that's different from what I thought we were saying. I thought we were saying we'd just be replacing the `HarmonicAngleForce` tag with an `AngleForce` tag and whether it is harmonic or not would be specified by an attribute of the `SMIRFF` tag, as per what I wrote in the new docs:

> ... Selection of the type of angle force applied would be handled in a similar manner, via an `AngleForce="harmonic"` tag or similar. [This feature is being planned.](https://github.com/open-forcefield-group/smarty/issues/179)

**Comment by jchodera** [comment]
Actually, we may want to split off the SMIRFF forcefield stuff to a new repo before doing that.

**Comment by davidlmobley** [comment]
@jchodera - regarding splitting into separate repo, yes, that was what we discussed by phone the other day. I made an issue and assigned it to you, but let me know if you want me to do it instead. 
https://github.com/open-forcefield-group/smarty/issues/175

(Whoever does it needs to get travis going, etc., on the new repo, as well as connect it up to omnia).

**Comment by jchodera** [comment]
Yep, I've got it slated for the weekend. Do we need it earlier than that?

**Comment by davidlmobley** [comment]
No, weekend is fine. But I'd be happy to do it myself if it would free up time for you to do the property calculator for densities or GB solvation.

File: The-SMIRFF-force-field-format.md (hunk 1 above)
Comment (davidlmobley) [reply The-SMIRFF-force-field-format.md:17]: @jchodera - any further comments on this?

Comment (jchodera) [reply The-SMIRFF-force-field-format.md:17]: > Wait, hold on -- that's different from what I thought we were saying. I thought we were saying we'd just be replacing the HarmonicAngleForce tag with an AngleForce tag and whether it is harmonic or not would be specified by an attribute of the SMIRFF tag, as per what I wrote in the new docs:

We're in agreement here!  But it still says `HarmonicAngleForce` and not `AngleForce`. Same thing with `HarmonicBondForce`.

**Comment by jchodera** [comment]
I've fallen behind again here, but I realized I needed to [finish the GBSA PR](https://github.com/open-forcefield-group/smarty/pull/182) before peeling of the SMIRFF code to another repo.

**Comment by jchodera** [comment]
You should be good to rebase or merge from master and then we can merge this in.

File: The-SMIRFF-force-field-format.md (hunk 1 above)
Comment (davidlmobley) [reply The-SMIRFF-force-field-format.md:17]: Yes, totally agree. Just need to make the associated change in the code and then we can do that. I'll rebase/merge.

**Pull request merged**